src/vmm_unix.ml

```diff
@@ -209,12 +201,59 @@ type solo5_target = Spt | Hvt
 let solo5_image_target image =
   let* abi = Solo5_elftool.query_abi (owee_buf_of_cstruct image) in
   match abi.target with
-  | Solo5_elftool.Hvt -> Ok Hvt
-  | Solo5_elftool.Spt -> Ok Spt
+  | Solo5_elftool.Hvt -> Ok (Hvt, Int32.to_int abi.version)
+  | Solo5_elftool.Spt -> Ok (Spt, Int32.to_int abi.version)
   | x -> Error (`Msg (Fmt.str "unsupported solo5 target %a" Solo5_elftool.pp_abi_target x))
 
 let solo5_tender = function Spt -> "solo5-spt" | Hvt -> "solo5-hvt"
 
+let check_solo5_tender target version =
+  let cmd_names =
+    let base = solo5_tender target in
+    [ base ^ "." ^ string_of_int version ; base ]
+  in
+  let cmds =
+    List.concat_map (fun name ->
+        [ Bos.Cmd.(v (p Fpath.(!dbdir / name))) ; Bos.Cmd.v name ])
+      cmd_names
+  in
+  let* cmd =
+    Result.map_error
+      (fun _ ->
+         `Msg (Fmt.str "tender does not exist, looked for %a"
+                 Fmt.(list ~sep:(any ", ") string)
+                 (List.map Bos.Cmd.to_string cmds)))
+      (List.fold_left (fun acc name ->
+           match acc with
+           | Ok _ as cmd -> cmd
+           | Error _ ->
+             let db_pre = Fpath.(!dbdir / name) in
+             if Bos.OS.File.is_executable db_pre then
```

Comment:
we can use `Bos.OS.Cmd.must_exist` here, since if this gets the full path passed, it doesn't check for existance neither executability (as documented in the Bos.OS.Cmd moduile).